### PR TITLE
Implement WordPress API endpoint URL patterns

### PR DESCRIPTION
**Pull request by MizukiTemma**
### Short description

This PR contains the newly implemented URL patterns for WordPress API endpoint.

### Proposed changes

- New URL pattern was implemented.
-
-

### Resolved issues

Fixes: #855

### CODE REVIEW

File: src/api/urls.py [hunk 1]
```diff
@@ -135,3 +135,7 @@
         ),
     ),
 ]
+
+apiurl = [
+    url(r"^/wp-json/extensions/v3/$", include(urlpatterns)),
+]
```
Comment (melegiul) [inline src/api/urls.py:141]: According to the docs [(URLconf)](https://docs.djangoproject.com/en/3.1/topics/http/urls/#how-django-processes-a-request) django will load the url config and look for a variable called `urlpatterns`. So one option would be to append this list to `urlpatterns` sequence. But then there is the problem, that in our ROOT_URLCONF [API-urls](https://github.com/Integreat/integreat-cms/blob/e3b7c733822b3be05a5202b9b2e1a6c605e4e681/src/backend/urls.py#L25) all urls in this file get a preceeding `api` and this violates wordpress API specification.
Do you have an idea how to solve this problem?

**Review by melegiul — CHANGES_REQUESTED** [review]
Thank you very much for your PR :+1: 
Your idea is basically right to support the wordpress API
But there is a remaining problem that we have to address

File: src/api/urls.py (hunk 1 above)
Comment (timoludwig) [reply src/api/urls.py:141]: @melegiul Good catch! I suggest to move the "api" prefix to the [src/api/urls.py](https://github.com/Integreat/integreat-cms/blob/563c174ac36cde9e922aa2b3a31c245e9e1da93b/src/api/urls.py) file in the following way:

```
api_urlpatterns = [
    # All existing urlpatterns here
]

urlpatterns = [
    url(r"^api/", include(api_urlpatterns)),
    url(r"^wp-json/extensions/v3/", include(api_urlpatterns)),
]
```
Apart from that, there are also a few endoints which are named differently (e.g. "sites" vs "regions"), so for this cases, we'd need a new additional rules.

**Comment by melegiul** [comment]
That's an interesting problem that you encountered here, which is not you're fault
The problem is that our tests use [reverse](https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#reverse) for the urls of [views](https://github.com/Integreat/integreat-cms/blob/develop/src/cms/tests/views/api_view_test.py), but in the docs it says:
```
The reverse() function can reverse a large variety of regular expression patterns for URLs,
but not every possible one. The main restriction at the moment is that the pattern cannot contain
alternative choices using the vertical bar ("|") character.
```
So I could offer two options:
1. Revolve the regex and split it in two entries, one for "regions" and one for "sites", so that reversing in the test can succeed
2. We could open an issue to alter the test case setup to support vertical bar regex. In this case please comment out the critical test cases: "api_regions", "api_regions_live", "api_regions_hidden" in the file [https://github.com/Integreat/integreat-cms/blob/develop/src/cms/tests/views/api_view_test.py](https://github.com/Integreat/integreat-cms/blob/develop/src/cms/tests/views/api_view_test.py) and add a comment: 
```
# TODO: find way to test API without using reverse
```
I have no preference between these options, in my opinion you can choose which one you like

**Comment by melegiul** [comment]
@timoludwig What do you think, is there another option to solve this problem?

**Comment by timoludwig** [comment]
> @timoludwig What do you think, is there another option to solve this problem?

In this case, I think I'd suggest to introduce another variable and add the urls with the prefix `regions/` to the normal api endpoint and the legacy `sites/` only to the wordpress-api-patterns:

```
api_urlpatterns = [
    # All existing urlpatterns without the region-patterns here
]

region_api_urlpatterns = [
    url(r"^$", regions, name="api_regions"),
    url(r"^live/?$", liveregions, name="api_regions_live"),
    url(r"^hidden/?$", hiddenregions, name="api_regions_hidden"),
    url(r"^pushnew/?$", pushnew, name="api_pushnew"),
]

urlpatterns = [
    url(r"^api/", include(api_urlpatterns)),
    url(r"^api/regions/", include(region_api_urlpatterns)),
    url(r"^wp-json/extensions/v3/", include(api_urlpatterns)),
    url(r"^wp-json/extensions/v3/sites/", include(region_api_urlpatterns)),
]
```

**Review by timoludwig — APPROVED** [review]
Thanks! :+1:
I think this looks good and works as expected! :tada:

I combined all your commits into one again.

**Review by melegiul — APPROVED** [review]
Thank you :+1:

**Pull request merged**